inference-engine/tests/ngraph_functions/src/gather_nd.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <numeric>
+#include <vector>
+
+#include "ngraph_functions/builders.hpp"
+
+namespace ngraph {
+namespace builder {
+
+std::shared_ptr<Node> makeGatherND(
+        const ngraph::Output<Node>& dataNode,
+        const ngraph::Shape& indicesShape,
+        const element::Type& indicesType,
+        const std::size_t batchDims) {
+    const auto indices = [&] {
+        const auto& dataShape = dataNode.get_shape();
+        const auto indicesCount = std::accumulate(begin(indicesShape), prev(end(indicesShape)),
+                                                  1ull, std::multiplies<std::size_t>{});
+        const auto sliceRank = indicesShape.back();
+
+        const auto maxDim = *std::max_element(begin(dataShape), end(dataShape));
+        std::random_device rd;
+        std::mt19937 gen(rd());
+        std::uniform_int_distribution<int> distrib(1, maxDim);
```

Comment:
Can we reuse `generateVector` here? You will have 2 cycles: first will generate random values, second will normalize them (distrib(gen) % dataShape[dim + batchDims]). I just don't like the idea to spread randomizer specific implementation throughout the tests.

Reply:
Fixed